streams/test-utils/src/main/java/org/apache/kafka/streams/test/ConsumerRecordFactory.java

```diff
@@ -175,11 +178,32 @@ public void advanceTimeMs(final long advanceMs) {
             -1L,
             timestampMs,
             TimestampType.CREATE_TIME,
-            ConsumerRecord.NULL_CHECKSUM,
+            (long) ConsumerRecord.NULL_CHECKSUM,
             serializedKey == null ? 0 : serializedKey.length,
             serializedValue == null ? 0 : serializedValue.length,
             serializedKey,
-            serializedValue);
+            serializedValue,
+            headers);
+    }
+
+    /**
+     * Create a {@link ConsumerRecord} with default topic name and given topic, key, value, and timestamp.
+     * Does not auto advance internally tracked time.
+     *
+     * @param key the record key
+     * @param value the record value
+     * @param timestampMs the record timestamp
+     * @return the generated {@link ConsumerRecord}
+     */
+    public ConsumerRecord<byte[], byte[]> create(final String topicName,
```

Comment:
Could we update the KIP wiki for this function as well?

Reply:
Added

Reply:
Seems that you picked 3 overloads arbitrarily for header extension. If we only consider the overloads that do not take any lists as parameters, there are 8 overloads to be extended. Any reason why you picked only 3?

Reply:
Just added the ones required by new tests. I'm adding the missing ones :)